ros2controlcli/ros2controlcli/verb/list_hardware_components.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 ROS2-Control Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from controller_manager import list_hardware_components
+from controller_manager.spawner import bcolors
+
+from ros2cli.node.direct import add_arguments
+from ros2cli.node.strategy import NodeStrategy
+from ros2cli.verb import VerbExtension
+from ros2controlcli.api import add_controller_mgr_parsers
+
+
+class ListHardwareComponentsVerb(VerbExtension):
+    """Output the list of available hardware components."""
+
+    def add_arguments(self, parser, cli_name):
+        add_arguments(parser)
+        add_controller_mgr_parsers(parser)
+
+    def main(self, *, args):
+        with NodeStrategy(args) as node:
+            hardware_components = list_hardware_components(node, args.controller_manager)
+
+            for idx, component in enumerate(hardware_components.component):
+                print(f'Hardware Component {idx}')
+                print(f'\tname: {component.name}')
+                print(f'\ttype: {component.type}')
```

Comment:
Probably mostly personal choice but I'd try merging these many print statements into one with a larger format string. After all  f-strings are meant for helping keeping big prints structured and readable.

Reply:
print statements are now combined

Reply:
thx!